internal/config/config.go

```diff
@@ -0,0 +1,123 @@
+package config
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/readium/r2-streamer-go/internal/consts"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Config holds all configuration for our program
+type Config struct {
+	EnvName         string
+	BindAddr        net.IP
+	BindPort        uint
+	SentryDSN       string
+	CacheDSN        string
+	PublicationPath string
+	StaticPath      string
+	Origins         []string
+
+	LogFile   string
+	LogFormat string
+	LogLevel  string
+}
+
+// NewConfig creates a Config instance
+func NewConfig() *Config {
```

Comment:
For "bag of data" structs without much behavior, prefer using value instead of pointer semantics. This is a bit safer by preventing an instance from being modified by an object that doesn't "own" it.

```suggestion
func NewConfig() Config {
```

Reply:
Sure thing. I do have a tendency to unnecessarily use pointers in go, so feel free to correct those.